### PR TITLE
Updated loss calculation

### PR DESCRIPTION
**Pull request by AAshqar**
Loss now considers the different dimensions coming from the ground truth data and the mixture parameters + autocorrelation is now also used in the gaussian function

### CODE REVIEW

**Review by mohammadbashiri — COMMENTED** [review]
Thanks Abdallah. I think the return should look something like:
return self.oneDivTwoPI * torch.reciprocal(out_sigma ** 2)) * torch.reciprocal(torch.sqrt(1-out_corr**2)) * torch.exp(result)

**Review by mohammadbashiri — APPROVED** [review]
